overlord/hookstate/ctlcmd/get.go

```diff
@@ -245,12 +251,22 @@ func attributesTask(context *hookstate.Context) (*state.Task, error) {
 	defer context.Unlock()
 
 	if err := context.Get("attrs-task", &attrsTaskID); err != nil {
-		return nil, err
+		if err == state.ErrNoState {
+			// if attrs-task is not present, look for attributes in hook's own task
```

Comment:
This is now needed because when disconnect hooks are run as part of snap removal, we don't have a "central" task for each connection as we **only** run disconnect hooks, so we don't have a per-connection task to store connection attributes so that that `snap ctl get` can read them. So instead I just store interface attributes on every disconnect hook and have this fallback here (as this code handles connect as well).

Connect doesn't have this problem, because whether with snap install/refresh we not only run connect hooks but also the regular "connect" task, which is also the place we store per-connection attributes at, and we point at it via "attrs-task".
 
We could get rid of this discrepancy between connect and disconnect and remove this fallback if we also run the disconnect task, however then we would need to think what to do about "discard-conns" task on snap removal, which as you noticed removes all connections "en masse".